### PR TITLE
feat: lookup cost center by service

### PR DESCRIPTION
**Pull request by jniles**
This commit adds supporting functions to look up the cost centers by the service UUID.  It also adds an integration test for this feature.

### CODE REVIEW

File: server/controllers/finance/accounts/index.js
Comment (jmcameron) [inline server/controllers/finance/accounts/index.js:418]: It looks like you only exported one function here.  But you exported both for lookup by service.  Was there a reason for the difference?

File: server/models/procedures/cost_centers.sql
Comment (jmcameron) [inline server/models/procedures/cost_centers.sql:38]: Why do we need a SQL function for this?  Wouldn't a normal server db query be fine for something this simple?  (like the ones in server/controllers/*/*.js)?

**Review by jmcameron — COMMENTED** [review]
The code looks good.  I had a couple of questions but nothing that should block this PR. 
I tried it on my system and the unit tests and integration test all ran fine.

However, I suggest you wait until we get the Fee Center -> Cost Center rename PR released before doing final adjustments so it can be released.

File: server/models/procedures/cost_centers.sql
Comment (jniles) [reply server/models/procedures/cost_centers.sql:38]: Two reasons:

1.  The invoicing and stock movements data calculation are doing in a stored procedure.  We've left JS at that point - it is all in SQL.  So, we need this to be available in SQL.

2. Making it a function instead of a straight query means we can change it easier when we rename the tables or change the way cost centers are linked to accounts/services.  For example, we may eventually link cost centers directly to accounts.

File: server/controllers/finance/accounts/index.js
Comment (jniles) [reply server/controllers/finance/accounts/index.js:418]: No, not really.  I just didn't need one of them yet.  Eventually, I'll probably export both!

**Comment by jniles** [comment]
@jmcameron this is ready for a second review.

File: server/models/procedures/cost_centers.sql
Comment (jmcameron) [reply server/models/procedures/cost_centers.sql:38]: ok

**Review by jmcameron — CHANGES_REQUESTED** [review]
I tested this on my computer.   All the tests passed except for the end-to-end tests.    Before the end-to-end tests would run, I had to make a couple of naming fixes (related to the Fee Center -> Cost Center renaming).  See the diffs file:
[pr5913.diffs.txt](https://github.com/IMA-WorldHealth/bhima/files/7184848/pr5913.diffs.txt)

When I ran the end-to-end tests, I saw this error:
```
  Distribution Auxiliary Cost Center
    1) set allocation by percentage

  34 passing (45s)
  1 pending
  1 failing

  1) Distribution Auxiliary Cost Center
       set allocation by percentage:
     NoSuchElementError: No element found using locator: By(css selector, *[id="allocation-center-grid"])
```
I presume this is related to the Fee Center -> Cost Center / Distribution -> Allocation renaming now merged into master.

**Comment by jniles** [comment]
@jmcameron I think this happens for me on `master`.  My end to end tests haven't worked for some time.  I don't think it's worth fixing something we are going to be rewriting imminently, right?

**Comment by jmcameron** [comment]
> @jmcameron I think this happens for me on `master`. My end to end tests haven't worked for some time. I don't think it's worth fixing something we are going to be rewriting imminently, right?

Ok.  I still suggest you go ahead and apply the diffs before releasing so that the end-to-end tests actually start (that way we fix these on master).  We can deal with actually fixing the end-to-end tests later when the dust has settled.

**Comment by jmcameron** [comment]
> Ok. I still suggest you go ahead and apply the diffs before releasing so that the end-to-end tests actually start (that way we fix these on master). We can deal with actually fixing the end-to-end tests later when the dust has settled.

Or we could do it on a separate PR?

**Comment by jniles** [comment]
@jmcameron changes landed.

**Pull request merged**
